### ISSUE TITLE
Detection not working for files containing special characters

### ISSUE THREAD

**Issue by ReneBL** [microsoft/ApplicationInspector#601]
**Describe the bug**

Given some valid ApplicationInspector rule, the detection doesn't work properly in files containing special characters. For example, in the context of a Maven project containing files [using the Maven Resources Plugin filtering feature](https://maven.apache.org/plugins/maven-resources-plugin/examples/filter.html), specific delimiters like `${...}` or `@...@` can be used

Quoted from the linked docs:

> Variables can be included in your resources. These variables, denoted by the ${...} or @...@ delimiters, can come from the system properties, your project properties, from your filter resources and from the command line.

The problem is that ApplicationInspector won't detect any match from any rule that applies to files containing such characters

**To Reproduce**
1. Define a yaml file like

```yaml
   app:
      version: @project.version@
   myproject:
      config: xyz
```

2. Define a rule detecting the `xyz` value of the `myproject.config` property
3. Run ApplicationInspector with `-s . -f json --processing-timeout 900000 --file-timeout 300000 -A -u --no-show-progress -r /rules -i -M 20 -o result.json`
4. Open up the analysis result file
5. The tag associated to the defined rule is not detected

**Expected behavior**
The rule defined in step 2 should have detected a match in the file presented in step 1

**Operating Environment (please complete the following information):**
- Application Inspector Version: 1.9.22
- OS: Linux Ubuntu 20.04

**Comment by gfs** [comment]
Can you please share a sample of the rule (or a example rule that demonstrates the behavior) to reproduce with?

**Comment by gfs** [comment]
The issue is that this file is not valid yaml. 

Per the Yaml spec https://yaml.org/spec/1.2.2/#23-scalars, see example 5.10 - scalar values cannot start with `@`. 

YamlDotNet, which is the dependency we use to parse Yaml, enforces this, as is reflected by the error returned by Application Inspector
 
![Image](https://github.com/user-attachments/assets/83684cca-50c4-4725-8f25-6ac4166be5ac)

I don't see a path forward to parsing such files using YamlPaths unless YamlDotNet decided to go off spec and allow those special characters to start scalar values, so to use Application Inspector on these files you'd either have to perform the preprocessing to get rid of the Maven variable components, or switch to simple regex.
